ui/src/shared/components/inlineLabels/InlineLabelsEditor.tsx

```diff
@@ -0,0 +1,231 @@
+// Libraries
+import React, {Component, ChangeEvent} from 'react'
+import _ from 'lodash'
+
+// APIs
+import {client} from 'src/utils/api'
+
+// Components
+import {
+  Button,
+  ButtonShape,
+  IconFont,
+  ComponentColor,
+} from '@influxdata/clockface'
+import InlineLabelPopover from 'src/shared/components/inlineLabels/InlineLabelPopover'
+import CreateLabelOverlay from 'src/configuration/components/CreateLabelOverlay'
+
+// Types
+import {Label} from '@influxdata/influx'
+
+// Utils
+import {validateLabelUniqueness} from 'src/configuration/utils/labels'
+
+// Styles
+import 'src/shared/components/inlineLabels/InlineLabelsEditor.scss'
+
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+enum OverlayState {
```

Comment:
there might (i think) be an overlay state enum already defined elsewhere that could be used

Reply:
I think you are right